### PR TITLE
What is your age

### LINKED ISSUE #8: Write a method to calculate the age of a person
Input: Date of birth
Calculate age in years

Resources:
[C# DateTime Properties](https://docs.microsoft.com/en-us/dotnet/api/system.datetime?view=netcore-3.1#properties)

### PR DESCRIPTION
**Pull request by drtaupier**
image

### CODE REVIEW

File: CSharpBasics/BasicProblems.cs [hunk 1]
```diff
@@ -21,8 +21,8 @@ public static void SumTwoNumbers()
             int.TryParse(userInput, out int num2);
 
             //Display result
-            var result = num1 + num2; 
-            Console.WriteLine($"The result is: {result}" );
```
Comment (mriverap) [inline CSharpBasics/BasicProblems.cs:25]: This belongs to another PR.

File: CSharpBasics/BasicProblems.cs [hunk 2]
```diff
@@ -21,8 +21,8 @@ public static void SumTwoNumbers()
...
+            var result = num1 + num2;
+            Console.WriteLine($"The result is: {result}");
         }
         public static void MultiplyTwoNumbers()
```
Comment (mriverap) [inline CSharpBasics/BasicProblems.cs:27]: This belongs to another PR

File: CSharpBasics/BasicProblems.cs [hunk 3]
```diff
@@ -77,6 +79,54 @@ public static void IsEvenOrOdd()
             Console.Write("Please, enter a number: ");
             var inputNumber = Console.ReadLine();
             int.TryParse(inputNumber, out int num1);
+
+            if (num1 % 2 == 0)
```
Comment (mriverap) [inline CSharpBasics/BasicProblems.cs:83]: All this belongs to another PR

File: CSharpBasics/BasicProblems.cs [hunk 4]
```diff
@@ -77,6 +79,54 @@ public static void IsEvenOrOdd()
...
+            DateTime today = DateTime.Today;
+            //Get year
+            int yearNow = today.Year;
+            //Get month
+            int monthNow = today.Month;
+            //Get day
+            int dayNow = today.Day;
+            //subtracting years
+            var age = yearNow - yearOfBirth;
+            if (monthOfBirth > monthNow)
```
Comment (mriverap) [inline CSharpBasics/BasicProblems.cs]: Very good. You can combine the two "if" statements into one.

**Review by mriverap — CHANGES_REQUESTED** [review]
Please include only changes that belong to this problem.

File: CSharpBasics/BasicProblems.cs (hunk 4 above)
Comment (drtaupier) [reply CSharpBasics/BasicProblems.cs]: Done prima. now its less code 👍

File: CSharpBasics/BasicProblems.cs [hunk 5]
```diff
@@ -77,6 +79,67 @@ public static void IsEvenOrOdd()
...
+            //
+            double sum = 0;
+            for(var i = 1; i<=times; i++)
+            {
+                Console.Write("Please, enter a number: ");
+                var numberCase = Console.ReadLine();
+                double.TryParse(numberCase, out double num);
+                sum += num;
+            }
+            Console.WriteLine($"The result is: {sum / times}");
```
Comment (mriverap) [inline CSharpBasics/BasicProblems.cs:142]: This belongs in a different PR.

File: CSharpBasics/BasicProblems.cs [hunk 6]
```diff
@@ -77,6 +79,67 @@ public static void IsEvenOrOdd()
...
+            DateTime today = DateTime.Today;
+            //Get year
+            int yearNow = today.Year;
+            //Get month
+            int monthNow = today.Month;
+            //Get day
+            int dayNow = today.Day;
+            //subtracting years
+            var age = yearNow - yearOfBirth;
+            if (monthOfBirth > monthNow || monthOfBirth == monthNow && dayOfBirth > dayNow)
```
Comment (mriverap) [inline CSharpBasics/BasicProblems.cs:118]: This is the idea, but there is a potential error here. Did you check if it works as expected?

File: CSharpBasics/Program.cs [hunk 7]
```diff
@@ -27,7 +27,19 @@ static void Main(string[] args)
...
+                            BasicProblems.IsEvenOrOdd();
+                            break;
+                        case 6:
+                            BasicProblems.WhatIsYourAge();
+                            break;
+                        case 7:
+                            BasicProblems.WhatIsYourAge();
+                            break;
+                        case 8:
+                            BasicProblems.average();
```
Comment (mriverap) [inline CSharpBasics/Program.cs:42]: This is for a different PR.

**Review by mriverap — APPROVED** [review]

**Pull request merged**